device/Program.cs

```diff
@@ -0,0 +1,73 @@
+﻿using Microsoft.Azure.Amqp.Framing;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace device
+{
+    class Program
+    {
+        static bool sendModelAtConnection = true;
+        static bool sendModelAsProperties = true;
+        static string cs = System.Environment.GetEnvironmentVariable("DEVICE_CS");
+        static string model = File.ReadAllText(@"..\..\..\deviceModel.json");
+
+        static async Task Main(string[] args)
+        {
+            string selfDescribingDtmi = "dtmi:azure:common:SelfDescribing;1";
+            string targetModelId = "dtmi:com:example:myDevice;1";
+            string targetModelhash = common.Hash.GetHashString(model);
+
+            string modelId = selfDescribingDtmi;
+
+            if (sendModelAtConnection)
+            {
+                Uri u = new($"{selfDescribingDtmi}?tmhash={targetModelhash}&tmid={targetModelId}");
```

Comment:
What is the purpose of including these in the query string. Is that one proposal for self-describing design?

Reply:
it will allow consumers of the `deviceTwin.modelId` to verify the content returned by the device. Yes, this is part of the design.

Reply:
We have the same thing in the model, reported as properties. Is that an alternative we're considering? Are we doing both?